parsers/TH.py

```diff
@@ -154,23 +156,49 @@ def fetch_price(
     target_datetime: Optional[dt.datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> dict:
-    """Fetch the price data from the MEA."""
+    """
+    Fetch the base tariff data from the MEA (Unit in THB). This is then added up with
+    Float Time (Ft) rate from another MEA's webpage (Unit in Satang (THB/100)).
+
+    The Thai MEA/PEA Electicity Fee calculation are as follows:
+    AmountDue = (BasePrice** + FtRate) * UnitAmountInKWh
+    ActualDue = (AmountDue * 7%VAT) + FixedServiceFee
+
+    * Time-of-Use (TOU) rate is uncommon and thus left unimplemented.
+    **While actual BasePrice is done in a progressive manner, For Electricity Maps -
+      we use "AmountDue" at 1MWh calculated at the highest pricing bracket for simplification.
+    """
     if target_datetime is not None:
         raise NotImplementedError("This parser is not yet able to parse past dates")
 
     # Fetch price from MEA table.
-    with requests.get(MEA_PRICE_URL) as response:
-        soup = BeautifulSoup(response.content, "lxml")
+    # `price_base` is 'Over 400 kWh (up from 401st)' from Table 1.1
+    with requests.get(MEA_BASEPRICE_URL) as response:
+        soup_base = BeautifulSoup(response.content, "lxml")
+
+    unit_price_table = soup_base.find_all("table")[1]
+    price_base = unit_price_table.find_all("td")[19].text
+
+    # Available Ft pricing history dated back as far as September 2535 B.E. (1992 C.E.)
+    # `price_ft` slot's is 0+(month number), additional +13 is needed if that slot is " "
+    # For 2023 multi-Ft rates, we assumed the household rate.
+    with requests.get(MEA_FT_URL) as response:
+        soup_ft = BeautifulSoup(response.content, "lxml")
+
+    ft_rate_table = soup_ft.find_all("table")[1]
+    curr_ft_month = arrow.now(TZ).month
+    price_ft = ft_rate_table.find_all("td")[curr_ft_month].text
 
-    # 'Over 400 kWh (up from 401st)' from Table 1.1
-    unit_price_table = soup.find_all("table")[1]
-    price = unit_price_table.find_all("td")[19]
+    if price_ft == "\xa0":
+        price_ft = ft_rate_table.find_all("td")[curr_ft_month + 13]
+    if "\n" in price_ft:
+        price_ft = re.findall("\d+\.\d+", price_ft)[0]
 
     return {
         "zoneKey": zone_key,
         "currency": "THB",
         "datetime": arrow.now(TZ).datetime,
-        "price": float(price.text) * 1000,
+        "price": float(price_base) * 1000 + float(price_ft) * 10,
```

Comment:
Should this not be 100 based on the above comments?

Reply:
The `base_price` for May 2023 outputs as 4.4217 THB

The `ft_price` we got from another MEA's webpage on the other hand would be 91.19 Satang or 0.9119 THB - so rather than divide it by 100 just to remultiply it by 1,000 again later to match `base_price`, we can just multiply it by 10 for 1MWh calculation. :)

Reply:
Sounds good!